### DISCUSSION TITLE
Problema para cargar las tareas

### DISCUSSION THREAD

**Discussion by santiago-collab** [rramosp/ai4eng.v1#9] [02 - PYTHON]
Buenas noches, profesor y todos.

Tenía una pregunta y algo que reportar, desde hace unos días estoy tratando de enviar las actividades correspondientes al LAB 02.03 - PANDAS, sin embargo, cuando ejecuto el codigo [student.submit_task(globals(), task_id="task_02");], me genera el siguiente error:
error: ERROR EXECUTING YOUR CODE
---------------------------------
Traceback (most recent call last):
  File "calling_script.py", line 24, in <module>
    r = eval("ofawdqko.{}(source_functions, source_variables, 'santiago_200012@hotmail.com')".format(grader_function_name), namespace)
  File "<string>", line 1, in <module>
  File "/content/ofawdqko.py", line 46, in grader_02
    answer = st_get_stats(d)
  File "<string>", line 4, in get_stats
NameError: name 'd' is not defined
 (set session.debug=True for tracebacks)

Ya revisé todo lo relacionada al correo y contraseña de registro y ambas cosas están bien. Les agradecería mucho su ayuda con este problema o sus sugerencias. Muchas gracias, quedo atento.

**Comment by rramosp** [comment]
hola santiago, 

es raro porque llevo un par de semestres ya usando el corrector de este task y por ahora ha funcionado bien.

pero quien sabe, quizá has dado con un caso de uso particular.

mándame porfa tu solución al correo para poder yo verificar el corrector con la misma: raul.ramos@udea.edu.co

saludos
raúl

**Reply by santiago-collab** [reply]
Buenos días profe Raúl. 
Ya se solucionó el problema. Al parecer el error sale cuando hay celdas adicionales a las que pide el notebook. En mi caso, tenía unas celdas más para hacer algunas comprobaciones y ensayos antes de hacer el código a entregar. 

Muchas gracias.

**Reply by Astolfo2332** [reply]
Hola profe tengo el mismo problema que el compañero me sale: 
```
RLXMOOCException                          Traceback (most recent call last)
[<ipython-input-67-fde445fafb4e>](https://localhost:8080/#) in <module>
----> 1 student.submit_task(globals(), task_id="task_02");

1 frames
/content/local/lib/rlxmoocapi/session.py in submit_task(self, namespace, course_id, session_id, lab_id, task_id, user_id, display_html)

RLXMOOCException: error: ERROR EXECUTING YOUR CODE
---------------------------------
Traceback (most recent call last):
  File "calling_script.py", line 24, in <module>
    r = eval("qxyeoejc.{}(source_functions, source_variables, '[miguel.lopez4@udea.edu.co](mailto:miguel.lopez4@udea.edu.co)')".format(grader_function_name), namespace)
  File "<string>", line 1, in <module>
  File "/content/qxyeoejc.py", line 46, in grader_02
    answer = st_get_stats(d)
  File "<string>", line 7, in get_stats
NameError: name 'd' is not defined
 (set session.debug=False to remove tracebacks)
```

En mi caso no tengo celdas adicionales de las que pide el notebook, qué puedo hacer? 

Edit: el mismo error me sale en la task 3

**Reply by Astolfo2332** [reply]
Bueno, lo siento por las molestias pero el error es bastante simple, en efecto la variable "d" no existe debido a que la función que uno crea tiene como argumento "df" , así que cuando uno corre el programa y hace las pruebas este toma la variable "d" anteriormente creada y que existe en el entorno pero obviamente cuando uno manda la respuesta al calificador dicha variable no existe. Así que como amigable recordatorio estén pendientes del nombre de las variables y espero que esto le sirva a alguien.

TL;NR la variable "d" no existe en el calificador solo en tu entorno revisar el nombre de las variables en la función.

**Comment by rramosp** [comment]
gracias Miguel ... buen punto ... efectivamente, los correctores
automáticos solo "ven" lo que hay dentro de la función del ejercicio... no
"ven" nada del resto del notebook ... por tanto si tu función depende de
una variable que esté en el notebook pero no en la función fallará al ser
ejecutada por el corrector

On Sat, Feb 11, 2023 at 4:44 PM Miguel ***@***.***> wrote:

> Bueno, lo siento por las molestias pero el error es bastante simple, en
> efecto la variable "d" no existe debido a que la función que uno crea tiene
> como argumento "df" , así que cuando uno corre el programa y hace las
> pruebas este toma la variable "d" anteriormente creada y que existe en el
> entorno pero obviamente cuando uno manda la respuesta al calificador dicha
> variable no existe. Así que como amigable recordatorio estén pendientes del
> nombre de las variables y espero que esto le sirva a alguien.
>
> TL;NR la variable "d" no existe en el calificador solo en tu entorno
> revisar el nombre de las variables en la función.
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/rramosp/ai4eng.v1/discussions/9#discussioncomment-4945984>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/ADJRZIIPGCG2ZSBXQLYUZPDWXAB2FANCNFSM44OH6HZQ>
> .
> You are receiving this because you commented.Message ID: <rramosp/ai4eng.
> ***@***.***>
>

-- 
*Aviso legal:* El contenido de este mensaje y los archivos adjuntos son 
confidenciales y de uso exclusivo de la Universidad Nacional de Colombia. 
Se encuentran dirigidos sólo para el uso del destinatario al cual van 
enviados. La reproducción, lectura y/o copia se encuentran prohibidas a 
cualquier persona diferente a este y puede ser ilegal. Si usted lo ha 
recibido por error, infórmenos y elimínelo de su correo. Los Datos 
Personales serán tratados conforme a la Ley 1581 de 2012 y a nuestra 
Política de Datos Personales que podrá consultar en la página web 
www.unal.edu.co <http://www.unal.edu.co/>.* *Las opiniones, informaciones, 
conclusiones y cualquier otro tipo de dato contenido en este correo 
electrónico, no relacionados con la actividad de la Universidad Nacional de 
Colombia, se entenderá como personales y de ninguna manera son avaladas por 
la Universidad.
